### PR TITLE
refactor: add make case classes final

### PR DESCRIPTION
**Pull request by tusharmath**
fixes: Unable to generate ZIO Schema for the ADT.

### CODE REVIEW

**Comment by ghostdogpr** [comment]
How about just doing it for the ADTs? We don't need it in tests and it make the code longer.

**Comment by tusharmath** [comment]
Made the changes.

**Comment by tusharmath** [comment]
Will this be automatically publish in the snapshot?

File: core/src/main/scala/caliban/Value.scala [hunk 1]
```diff
@@ -149,19 +149,19 @@ object Value {
...
       override def toFloat: Float           = value.toFloat
       override def toDouble: Double         = value
       override def toBigDecimal: BigDecimal = BigDecimal(value)
       override def toString: String         = value.toString
     }
-    case class BigDecimalNumber(value: BigDecimal) extends FloatValue {
+    final case class BigDecimalNumber(value: BigDecimal) extends FloatValue {
       override def toFloat: Float           = value.toFloat
       override def toDouble: Double         = value.toDouble
       override def toBigDecimal: BigDecimal = value
```
Comment (tusharmath) [inline core/src/main/scala/caliban/Value.scala:167]: This was required for ZIO Schema to encode `Document`

**Review by ghostdogpr — APPROVED** [review]

**Comment by ghostdogpr** [comment]
Yep, a snapshot will be created in a few minutes.

**Pull request merged**